src/agent/block_store_services/block_store_client.js

```diff
@@ -120,52 +119,51 @@ class BlockStoreClient {
         const { timeout = config.IO_WRITE_BLOCK_TIMEOUT } = options;
         const { block_md } = params;
         const data = params[RPC_BUFFERS].data;
+        let usage;
 
         return P.resolve()
             .then(async () => {
-
-                const {
-                    usage,
-                    signed_url,
-                    disable_delegation,
-                    // disable_metadata,
-                    s3_params,
-                    write_params,
-                } = await rpc_client.block_store.delegate_write_block({ block_md, data_length: data.length }, options);
-
                 try {
-                    if (disable_delegation) {
-                        if (!s3_params || !write_params) {
-                            throw new Error('expected delegate_write_block to return write_params');
-                        }
-                        dbg.log1('got s3_params from block_store. writing using S3 sdk. s3_params =',
-                            _.omit(s3_params, 'secretAccessKey'));
-                        s3_params.httpOptions = _.omitBy({ agent: http_utils.get_unsecured_http_agent(s3_params.endpoint) },
-                            _.isUndefined);
-                        const s3 = new AWS.S3(s3_params);
-                        write_params.Body = data;
-                        await s3.putObject(write_params).promise();
-                    } else {
-                        const req_options = {
-                            url: signed_url,
-                            method: 'PUT',
-                            followAllRedirects: true,
-                            body: data,
-                            rejectUnauthorized: http_utils.should_reject_unauthorized(signed_url)
-                        };
-
-                        const res = await P.fromCallback(callback => request(req_options, callback));
-
-                        // if not OK parse the error and throw Error object
-                        if (res.statusCode !== 200) {
-                            await this._throw_s3_err(res);
-                        }
-
+                    if (!this.block_to_creds) {
```

Comment:
This is a map that never expires....
We should change this to use `LRUCache`.

Reply:
does `LRUCache` really fit here? the block_to_creds also holds the usage counter, so before any invalidation, there is a need to send it to the block store, which makes it a bit more complicated. we don't expect to have too many cloud resources

Reply:
LRUCache for the creds, simple map for io_stats